tuf/repository_lib.py

```diff
@@ -131,10 +131,27 @@ def _generate_and_write_metadata(rolename, metadata_filename,
 
 
   elif rolename == 'snapshot':
-    metadata = generate_snapshot_metadata(metadata_directory,
-        roleinfo['version'], roleinfo['expires'],
-        storage_backend, consistent_snapshot, repository_name,
-        use_length=use_snapshot_length, use_hashes=use_snapshot_hashes)
+    if (snapshot_merkle):
```

Comment:
```suggestion
    if snapshot_merkle:
```
No need for braces around a conditional in Python